src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,40 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupForm.querySelector(
+      ".form__button_type_delete"
+    );
+    // this._submitButtonText = this._submitButton.textContent;
+  }
+
+  setSubmitAction(action) {
+    console.log("aaa");
+    this._handleFormSubmit = action;
+  }
+
+  setLoading(isLoading, loadingText = "Removing...") {
+    if (isLoading) {
+      this._submitButton.textContent = loadingText;
+      this._submitButton.disabled = true;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+      this._submitButton.disabled = false;
+    }
+  }
+
+  close() {
+    super.close();
+    this._popupForm.removeEventListener("submit", this._handleFormSubmit);
+  }
+
+  setEventListeners() {
```

Comment:
`setEventListeners` should be `_setEventListeners` (with a **low dash**) like the parent's method to let it be called in method `open`